lib/Parse/ParseIfConfig.cpp

```diff
@@ -623,8 +623,18 @@ ParserResult<IfConfigDecl> Parser::parseIfConfig(
     foundActive |= isActive;
 
     if (!Tok.isAtStartOfLine() && Tok.isNot(tok::eof)) {
-      diagnose(Tok.getLoc(),
-               diag::extra_tokens_conditional_compilation_directive);
+      if (isElse && Tok.is(tok::kw_if)) {
+        diagnose(Tok.getLoc(),
+                 diag::unexpected_if_following_else_compilation_directive)
+            .fixItReplace(
+                SourceRange(ClauseLoc, Tok.getLoc().getAdvancedLoc(1)),
```

Comment:
Maybe you can extract this into a separate line otherwise it makes the indentation looks a bit weird IMO.

```c++
auto diag = diag::unexpected_if_following_else_compilation_directive;
auto rangeToReplace = SourceRange(ClauseLoc, Tok.getLoc().getAdvancedLoc(1));
diagnose(Tok.getLoc(), diag).fixItReplace(rangeToReplace, "#elseif");
```

Reply:
Turns out this was a little more complicated then it needed to be, it reads much better now that I've simplified it a bit